gravitee-plugin-annotation-processors/README.adoc

```diff
@@ -0,0 +1,164 @@
+
+[source, xml]
+----
+<dependency>
+    <groupId>org.hibernate.validator</groupId>
+    <artifactId>hibernate-validator</artifactId>
+    <version>8.0.0.Final</version>
+</dependency>
+<dependency>
+    <groupId>org.glassfish.expressly</groupId>
+    <artifactId>expressly</artifactId>
+    <version>5.0.0</version>
+</dependency>
+----
+[start=3]
+. Add this plugin in your pom.xml to defined the generated sources as source file (so you can use them in your classes in your IDE)
+
+[source, xml]
+----
+<plugin>
+    <groupId>org.codehaus.mojo</groupId>
+    <artifactId>build-helper-maven-plugin</artifactId>
+    <version>1.7</version>
+    <executions>
+        <execution>
+            <id>add-source</id>
+            <phase>generate-sources</phase>
+            <goals>
+                <goal>add-source</goal>
+            </goals>
+            <configuration>
+                <sources>
+                    <source>generated-sources</source>
+                </sources>
+            </configuration>
+        </execution>
+    </executions>
+</plugin>
+----
+[start=4]
+. Add the annotation `@ConfigurationEvaluator` at the class level of your plugin configuration classes (ie: KafkaEndpointConnectorConfiguration, KafkaEndpointConnectorSharedConfiguration) with the right value for the parameter `attributePrefix`
+. Add the validation annotations in your configuration classes as described in <<_validate_the_dynamic_configuration>>
+. Build your project to generate the new "ConfigurationEvaluator" classes (you can find them in the folder target/generated-sources/annotations)
+. Create all `evaluators` necessary in your constructor (only one attribute is required by the evaluator constructor which is the configuration bean) and call the `eval` method to generate the dynamic configuration object or retrieved it from the context if it has already been called once (it is stored in the context using an internal attribute to avoid multiple evaluation). Here you have an example for an endpoint :
+
+[source, java]
+----
+    // parts of the code are missing to focus on what needs to be changed
+
+    @Override
+    public Completable subscribe(final ExecutionContext ctx) {
+        return kafkaEndpointConnectorConfigurationEvaluator
+            .eval(ctx)
+            .flatMapCompletable(evaluatedConfiguration ->
+                kafkaEndpointConnectorSharedConfigurationEvaluator
+                    .eval(ctx)
+                    .flatMapCompletable(evaluatedSharedConfiguration ->
+                        Completable.fromRunnable(() -> {
+                            if (evaluatedSharedConfiguration.getConsumer().isEnabled()) {
+                                ctx
+                                    .response()
+                                    .messages(
+                                            // ...
+                                    );
+                            }
+                        })
+                    )
+            );
```

Comment:
We could probably simplify the example used in the readme by just using a single evaluator wdyt?

Reply:
We should also see how the evaluator should be instantiated (ie: in the constructor, using the original config).